src/update_engine/payload_processor.cc

```diff
@@ -145,35 +173,60 @@ ActionExitCode PayloadProcessor::LoadManifest() {
 ActionExitCode PayloadProcessor::PerformOperation() {
   DCHECK(next_operation_num_ < num_total_operations_);
 
-  const InstallOperation &op =
-      manifest_.partition_operations(next_operation_num_);
+  const size_t total_partition_operations = manifest_.partition_operations_size();
+  const InstallOperation *op = nullptr;
+  DeltaPerformer *performer = nullptr;
 
-  if (op.data_length() && op.data_offset() != buffer_offset_) {
+  if (next_operation_num_ < total_partition_operations) {
+    op = &manifest_.partition_operations(next_operation_num_);
+    performer = &partition_performer_;
+  } else {
+    size_t proc_operation = next_operation_num_ - total_partition_operations;
+
+    for (const InstallProcedure &proc : manifest_.procedures()) {
```

Comment:
this is a crazy way to index into an array; if this type gives you standard iterator semantics (and it should if c++11 for-in works?) it might be more readable.